magefiles/sprayproxy.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	routev1 "github.com/openshift/api/route/v1"
+	kubeCl "github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+const (
+	SPRAYPROXY_NAMESPACE = "sprayproxy"
+	ROUTE_NAME            = "sprayproxy-route"
+	PAC_NAMESPACE = "openshift-pipelines"
+	PAC_ROUTE_NAME = "pipelines-as-code-controller"
+)
+
+type SprayProxy struct {
+	BaseURL    string
+	Token      string
+	HTTPClient *http.Client
+}
+
+func NewSprayProxy(url string, token string) *SprayProxy {
+	return &SprayProxy{
+		BaseURL: url,
+		Token:   token,
+		HTTPClient: &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true, // #nosec G402
```

Comment:
also the #nosec should be added in the line before 
```suggestion
					// #nosec G402
					InsecureSkipVerify: true, 
```

Reply:
fixed